_profiles/drupal/resource-versioning/index.md

```diff
@@ -0,0 +1,273 @@
+(U+003A COLON, “:”) string. The first segment of the string **SHOULD** be
+interpreted as an identifier for a _version negotiation mechanism_. A version
+negotiation mechanism defines how a server will locate an appropriate resource
+version. Subsequent segments of the string **SHOULD** be interpreted as version
+negotiation arguments for the preceding mechanism. Collectively, this query
+parameter value is known as the _version identifier_.
+
+> Note: For example, a server may support both ID-based and time-based
+> mechanisms for requesting a resource version. The former mechanism would be
+> useful for comparing versions and the latter could be useful for requesting a
+> resource as it existed at an arbitrary point in time. This profile does not
+> attempt to define every possible mechanism for versioning resources.
+
+```
+                   version-identifier
+                   _______|_________
+                  /                \
+?resourceVersion=rel:latest-version
+                  \_/ \____________/
+                   |        |
+         version-negotiator |
+                     version-argument
+```
+
+## Server Responsibilities
+
+<a id="bad-version-negotiator"></a>A server **MUST** respond with `400 Bad Request` if a version negotiator is not
+supported. In this case, an error object that includes a `type` link to
+`https://jsonapi.org/profiles/drupal/resource-versioning/#bad-version-negotiator`
+**MUST** be included in the response document.
+   
+<a id="bad-version-argument"></a>If a server cannot process the given version argument for the given negotiation
+mechanism, it **MUST** respond with a `400 Bad Request`. In this case, an error
+object that includes a `type` link to
+`https://jsonapi.org/profiles/drupal/resource-versioning/#bad-version-argument`
+**MUST** be included in the response document.
+
+If a server is able to process the version argument but an appropriate version
+cannot be located, the server **MUST** respond with a `404 Not Found`.
+
+# Links
+
+When a server processes a request with a `resourceVersion` query parameter and
+a `self` link is provided for a top-level links object, the link's `href`
+**MUST** include the `resourceVersion` query parameter with the same version
+identifier that was requested.
+
+When a server processes a request with a `resourceVersion` query parameter
+all resource object `self` links **SHOULD** contain a `resourceVersion` query
+parameter which identifies the specific revision represented by that resource
+object.
+
+For example, in the following response document the `self` links are not the
+same:
+
+```json
+{
+  "data": {
+    "type": "article",
+    "id": 1,
+    "links": {
+      "self": "/article/1?resourceVersion=id:42"
+    }
+  },
+  "links": {
+    "self": "/article/1?resourceVersion=rel:latest-version"
+  }
+}
+```
+
+A server **MAY** provide a `version-history` link in a resource object's links
```

Comment:
Should `version-history` use the same version negotiator as the request? Should it always be `rel`?

Reply:
No. It should always be the most specific negotiator provided by the server.

Reply:
I should add that language instead of assuming it's implied.

Reply:
7a352dc